File: snow-chat-flutter/lib/core/constants/ws_cmd.dart
/// WebSocket 指令码，与后端 [WsCmd] 常量一一对应
/// WebSocket command codes, matching the backend WsCmd constants one-to-one.
class WsCmd {
  // -------------------------------------------------------------------------
  // 客户端 → 服务端（编号范围 1001–1099）
  // Client → Server (range 1001–1099)

  /// 用户登录
  /// User login
  static const int login = 1001;

  /// 发送文本消息
  /// Send text message
  static const int msgText = 1002;

  /// 发送图片消息
  /// Send image message
  static const int msgImage = 1003;

  /// 发送视频消息
  /// Send video message
  static const int msgVideo = 1004;

  /// 发送好友申请
  /// Send friend request
  static const int friendAddReq = 1006;

  /// 创建群组
  /// Create group
  static const int groupCreate = 1009;

  /// 消息已读回执
  /// Message read acknowledgment
  static const int readAck = 1012;

  /// 接收方确认收到消息（Client → Server）
  /// Receiver confirms message received (Client → Server)
  static const int msgReceipt = 1013;

  /// 进入聊天页面时请求未推送消息（Client → Server）
  /// Request undelivered messages when entering chat page (Client → Server)
  static const int fetchUndelivered = 1014;

  // -------------------------------------------------------------------------
  // 服务端 → 客户端（编号范围 2001–2099）
  // Server → Client (range 2001–2099)

  /// 服务器主动推送消息
  /// Server-initiated message push
  static const int msgPush = 2001;

  /// 好友申请通知
  /// Friend request notification
  static const int friendReqNotify = 2002;

  /// 在线状态变更
  /// Online status change
  static const int onlineStatus = 2003;

  /// 群组变动通知
  /// Group change notification
  static const int groupNotify = 2004;

  /// 消息发送成功回执
  /// Message sent acknowledgment
  static const int msgAck = 2005;

  /// 好友申请被接受
  /// Friend request accepted
  static const int friendAccepted = 2006;

  /// 服务器回执给发送方，确认已收到并推送（Server → Client）
  /// Server acknowledges receipt and push to sender (Server → Client)
  static const int msgReceiptAck = 2007;

  /// 服务器推送未推送成功的消息列表（Server → Client）
  /// Server pushes list of undelivered messages (Server → Client)
  static const int fetchUndeliveredAck = 2008;
}

File: snow-chat-im-backend/snow-im-api/src/main/java/com/stylesmile/chat/mqtt/WsCmd.java
package com.stylesmile.chat.mqtt;

/**
 * MQTT 消息命令常量，与 Flutter 端 ws_cmd.dart 保持一致
 */
public final class WsCmd {

    private WsCmd() {
    }

    // ==================== Client → Server (1001-1099) ====================
    /** 登录 */
    public static final int LOGIN = 1001;
    /** 文本消息 */
    public static final int MSG_TEXT = 1002;
    /** 图片消息 */
    public static final int MSG_IMAGE = 1003;
    /** 视频消息 */
    public static final int MSG_VIDEO = 1004;
    /** 好友添加请求 */
    public static final int FRIEND_ADD_REQ = 1006;
    /** 创建群组 */
    public static final int GROUP_CREATE = 1009;
    /** 已读回执 */
    public static final int READ_ACK = 1012;
    /** 接收方确认收到消息（Client→Server） */
    public static final int MSG_RECEIPT = 1013;
    /** 进入聊天页面时请求未推送消息（Client→Server） */
    public static final int FETCH_UNDELIVERED = 1014;

    // ==================== Server → Client (2001-2099) ====================
    /** 消息推送 */
    public static final int MSG_PUSH = 2001;
    /** 好友请求通知 */
    public static final int FRIEND_REQ_NOTIFY = 2002;
    /** 在线状态 */
    public static final int ONLINE_STATUS = 2003;
    /** 群组通知 */
    public static final int GROUP_NOTIFY = 2004;
    /** 消息确认 */
    public static final int MSG_ACK = 2005;
    /** 好友已接受 */
    public static final int FRIEND_ACCEPTED = 2006;
    /** 服务器回执给发送方确认已收到并推送（Server→Client） */
    public static final int MSG_RECEIPT_ACK = 2007;
    /** 服务器推送未推送成功的消息列表（Server→Client） */
    public static final int FETCH_UNDELIVERED_ACK = 2008;
}
